### ISSUE TITLE
Custom Blender Viewport Rendering

### ISSUE THREAD

**Issue by zicklag** [katharostech/arsenal#15]
**In order to** make it easier to preview my game and reduce development iteration time

**As a** game developer

**I want** to be able to have a live render preview for what my game's lights and materials will look like in the Blender viewport. Similar to how you can select the Eevee renderer and view it live in the viewport, except it will use our game's renderer and/or shaders.

---

This needs some investigation. Something to look at: [Blender GPU Shader Modules](https://docs.blender.org/api/blender2.8/gpu.html).

**Comment by termhn** [comment]
A possibility here *might* be to create a shared texture between the game and Blender, so the came could draw into it and then blender could draw that onto a textured quad or something using the GPU capabilities you linked, and then render the manipulation widgets and whatever else over top.

**Comment by zicklag** [comment]
Yeah, that would be cool. It sounds like it could be just feasible enough and definitely worth investigation.

**Comment by Raj2032** [comment]
I would love to see that take place.

**Comment by Raj2032** [comment]
@termhn 
> A possibility here _might_ be to create a shared texture between the game and Blender, so the came could draw into it and then blender could draw that onto a textured quad or something using the GPU capabilities you linked, and then render the manipulation widgets and whatever else over top.

Sorry I didn't quite understand that, so if in object mode you want the models to use some special texture on it right?

**Comment by termhn** [comment]
No. Basically the Arsenal engine would render the scene into an off-screen framebuffer that is shared with blender. Then Blender would draw this framebuffer onto its own viewport by just drawing a single rectangle across the whole screen. Finally, Blender would then draw the little manipulator tool thingies on top of that

**Comment by Raj2032** [comment]
@termhn I see mate that sounds great :)

**Comment by zicklag** [comment]
Issue moved to Taiga: https://tree.taiga.io/project/zicklag-arsenal/us/90
